src/main/java/io/github/syst3ms/skriptparser/lang/VariableString.java

```diff
@@ -140,10 +177,36 @@ public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContex
     public String toString(TriggerContext ctx) {
         if (simple)
             return (String) data[0];
+
+        boolean occasional = TagManager.getOccasionalState();
+        int tags = 1;
         var sb = new StringBuilder();
-        for (var o : data) {
+        for (int i = 0; i < data.length; i++) {
+            var o = data[i];
             if (o instanceof Expression) {
-                sb.append(TypeManager.toString((Object[]) ((Expression<?>) o).getValues(ctx)));
+                sb.append(TypeManager.toString(((Expression<?>) o).getValues(ctx)));
+            } else if (o instanceof Tag) {
+                if (((Tag) o).isOccasional() && !occasional) {
+                    tags++;
+                    continue;
+                }
+                int indexOf = CollectionUtils.ordinalIndexOf(Arrays.asList(data), Tag.class, tags);
+                if (indexOf == -1)
+                    indexOf = data.length;
+                var sb2 = new StringBuilder();
+                for (int i2 = i + 1; i2 < indexOf; i2++) {
```

Comment:
Truly a nitpick, but my personal convention is to use `i, j, k, ...` rather than `i, i2, i3, ...`